server/src-lib/Hasura/GraphQL/Validate/Types.hs

```diff
@@ -810,36 +789,28 @@ data QueryReusability
   = Reusable !ReusableVariableTypes
   | NotReusable
   deriving (Show, Eq)
-$(makePrisms ''QueryReusability)
-
-instance Semigroup QueryReusability where
-  Reusable a <> Reusable b = Reusable (a <> b)
-  _          <> _          = NotReusable
-instance Monoid QueryReusability where
-  mempty = Reusable mempty
```

Comment:
You say this “isn’t actually a monoid,” but that’s just not true; this *is* a monoid. It satisfies all the monoid laws. It wouldn’t be a monoid if we defined `mempty = NotReusable`, but we don’t. As-written, it’s a perfectly cromulent monoid.

Reply:
Yes, having taken a second look, you are right. So it makes sense to treat `QueryReusability` as a monoid.